packages/fxa-auth-server/lib/routes/account.js

```diff
@@ -1487,6 +1487,39 @@ module.exports = (
         return { subscriptions };
       },
     },
+    {
+      method: 'PUT',
+      path: '/account/metrics/ecosystemAnonId',
+      apidoc: {
+        errors: [error.invalidScopes],
+      },
+      options: {
+        auth: {
+          payload: false,
+          strategy: 'oauthToken',
+        },
+        validate: {
+          payload: {
+            ecosystemAnonId: isA.string().required(),
+          },
+        },
+      },
+      handler: async function (request) {
+        log.begin('account.updateEcosystemAnonId', request);
+
+        const { uid, scope } = request.auth.credentials;
+        const { ecosystemAnonId } = request.payload;
+        const scopeSet = ScopeSet.fromArray(scope);
+
+        if (!scopeSet.contains('profile:ecosystem_anon_id')) {
```

Comment:
Does this need to be `profile:ecosystem_anon_id:write` for write permissions?

Ref https://github.com/mozilla/fxa/blob/c5ab58b6ed6f37bd81a7172a7bbc9de52a82ccc0/packages/fxa-profile-server/docs/API.md#post-v1ecosystem_anon_id

Reply:
Drive-by since I saw this in my inbox - yes it does :-)